pkg/kb-summarizer/summarizer/summarizer.go

```diff
@@ -778,3 +802,89 @@ func printCheck(check *kb.Check) {
 func printCheckWrapper(cw *CheckWrapper) {
 	logrus.Debugf("checkWrapper: %+v", cw)
 }
+
+// handleAvMapData sets ActualValueMapData field for the fullReport and also set ActualValueNodeMap to nil for each CheckWrapper
+// in the report.
+func (s *Summarizer) handleAvMapData() error {
+
+	err := s.setFullReportActualValueMapData()
+	if err != nil {
+		return fmt.Errorf("failed to set actualValueMapData, err: %w", err)
+	}
+
+	// because of ActualValueNodeMap values the size of clusterscan report was exceeding the 1 MB limit for large clusters
+	// so this data is aggregated for all the checks and then set to ActualValueMapData field of the report after compression.
+	// and ActualValueNodeMap is set to nil for each check wrapper.
+	s.resetAvmapPerCheck()
+
+	return nil
+}
+
+// setFullReportActualValueMapData sets ActualValueMapData field for the fullReport
+func (s *Summarizer) setFullReportActualValueMapData() error {
+
+	avgroups := mapGroupWrappersToActualValueGroups(s.fullReport.GroupWrappers)
+
+	jsonData, err := json.Marshal(avgroups)
+	if err != nil {
+		return fmt.Errorf("Error encoding avgroups: %w", err)
```

Comment:
Errors are generally lowercase:
```suggestion
		return fmt.Errorf("error encoding avgroups: %w", err)
```